fiaas_deploy_daemon/deployer/kubernetes/pod_disruption_budget.py

```diff
@@ -52,7 +53,9 @@ def deploy(self, app_spec: AppSpec, selector: dict[str, any], labels: dict[str,
 
         # Conservative default to ensure that we don't block evictions.
         # See https://github.com/fiaas/fiaas-deploy-daemon/issues/220 for discussion.
-        max_unavailable = 1
+        max_unavailable = self.max_unavailable
+        if max_unavailable >= app_spec.autoscaler.max_replicas:
```

Comment:
You're right, and we should discuss how to manage the percentage case.

Reply:
One question, can we ignore this check if the max_unavailable is a string and has `%`? 😄

Reply:
I think the check is valid for the int value, and we need a different one (or no one) for the string value

Reply:
Also bear in mind we're already discarding the case of a single replica

Reply:
If we want to ensure the % allows evictions, I think something like this will work: 
`max_replicas - round(max_replicas*max_unavailable) >= min_replicas`
Let's say we have min_replicas=1, max_replicas=5, max_unavailable=80%. That means 5-(5*80%)=5-4=1 which still allows min_replicas, so we're good. A bigger % will leave us without pods, so we should disable PDB.
Looks good?

Reply:
I don't think any additional checks are needed if max_unavailable is a percentage. It should always be ok. But for integer values you want to ensure that max_unavailable is lower than min_replicas to avoid allowing all running pods to be evicted.

Reply:
> I think app_spec.autoscaler.min_replicas should also be checked here to avoid creating a PDB which doesn't allow any evictions (for example if config.pdb_max_unavailable >= min_replicas no evictions are allowed if the application is autoscaled to minimum).

I think I got this backwards in my previous (quoted) comment: The issue we try to avoid with ensuring config.pdb_max_unavailable < min_replicas and config.pdb_max_unavailable < max_replicas is not to avoid blocking evictions, but to avoid creating a PDB which allows evicting all replicas (defeating the purpose of the PDB in the first place).

config.pdb_max_unavailable is an int, it is a good idea to set max_unavailable=1 if config.pdb_max_unavailable >= min_replicas or config.pdb_max_unavailable >= max_replicas, to avoid creating a PDB that allows evicting all pods.

> One question, can we ignore this check if the max_unavailable is a string and has `%`? 😄

If config.pdb_max_unavailable is a str it could be set directly on the PDB without this check. If config.pdb_max_unavailable is set to 50% or less, I think it should result in requiring at least one available pod since replicas is at least 2 if we get to this logic. Maybe a good enough solution then is to set max_unavailable as is if it is a string, and document that values larger than 50% should be used with care? I suspect 50% should be high enough for most cases; what do you think?

Reply:
I think for risk that we can have a PDB with no effect, then max_replicas will need to be replaced with min_replicas (ie, the logic above is reversed). 

So we could have this instead, to ensure that we don't allow all replicas to be unavailable
if max_unavailable >= app_spec.autoscaler.min_replicas:
        max_unavailable = 1

Personally I think this is a bit hard to reason about, so I like the idea of allowing a percentage (like 10%) to be set for max_unavailable in the PDB.